src/ioq_server2.erl

```diff
@@ -108,26 +117,32 @@ call(Fd, Msg, Dimensions) ->
                 [couchdb, io_queue2, RW, bypassed_count]),
             gen_server:call(Fd, Msg, infinity);
         _ ->
-            DispatchStrategy = config:get(
-                "ioq2", "dispatch_strategy", ?DISPATCH_SERVER_PER_SCHEDULER),
-            Server = case DispatchStrategy of
-                ?DISPATCH_RANDOM ->
-                    SID = rand:uniform(erlang:system_info(schedulers)),
-                    ?SERVER_ID(SID);
-                ?DISPATCH_FD_HASH ->
-                    NumSchedulers = erlang:system_info(schedulers),
-                    SID = 1 + (erlang:phash2(Fd) rem NumSchedulers),
-                    ?SERVER_ID(SID);
-                ?DISPATCH_SINGLE_SERVER ->
-                    ?SERVER_ID(1);
-                _ ->
-                    SID = erlang:system_info(scheduler_id),
-                    ?SERVER_ID(SID)
-            end,
+            Server = ioq_server(Req, IOType),
             gen_server:call(Server, Req, infinity)
     end.
 
 
+ioq_server(#ioq_request{}, search) ->
+    ?IOQ2_SEARCH_SERVER;
+ioq_server(#ioq_request{fd=Fd}, _) ->
+    DispatchStrategy = config:get(
```

Comment:
I know this is copied from older code. One improvement I can immediately see (maybe subject to another PR).
Matching on atoms is faster than on strings.

```erlang
DispatchStrategy = list_to_atom(config:get("ioq2", "dispatch_strategy", ?DISPATCH_SERVER_PER_SCHEDULER),
case DispatchStrategy of
   ...
```

You would also would need to update defines to use atoms if you decide to go with this suggestion.

Reply:
The comparison is faster for sure, but we still need to convert the list from config to an atom and vice versa elsewhere, so always a bit of tradeoff when dealing with atoms and config strings, but I agree it would probably be better to compare against the atoms here.

Another awkward thing is that your code doesn't work, by switching `?DISPATCH_SERVER_PER_SCHEDULER` and the others to atoms, we can't use it as a default value in that config lookup, as it normally returns a string and so the fallback case would result in calling `list_to_atom` on an atom. There's no `config:get_atom`, to clean this up, so I think we would need to do:

```erlang
DispatchStrategy = case config:get("ioq2", "dispatch_strategy", undefined) of
    undefined -> ?DISPATCH_SERVER_PER_SCHEDULER;
    DispatchStrategy0 -> list_to_atom(DispatchStrategy0)
end,
```

Reply:
Like I said, atoms vs lists with config lookups is a bit awkward, but I've switched over to matching `?DISPATCH_STRATEGY` here against an atom in 2bb2f1c.